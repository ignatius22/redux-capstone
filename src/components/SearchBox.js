import React from 'react';
import PropTypes from 'prop-types';

const SearchBox = ({ searchChange }) => (
  <div className="pa2">
    <input
      data-testid="input"
      className="pa3 ba b--green bg-lightest-blue"
      type="search"
      placeholder="search your chicken"
      onChange={searchChange}
    />
  </div>
);

SearchBox.propTypes = {
  searchChange: PropTypes.func.isRequired,
};

export default SearchBox;
