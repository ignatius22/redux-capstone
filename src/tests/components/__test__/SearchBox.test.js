import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SearchBox from '../../../components/SearchBox';

it('it should render component', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchBox />, div);
});

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<SearchBox />).toJSON();
  expect(tree).toMatchSnapshot();
});
