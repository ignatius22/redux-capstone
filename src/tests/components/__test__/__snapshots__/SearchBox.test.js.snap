// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="pa2"
>
  <input
    className="pa3 ba b--green bg-lightest-blue"
    data-testid="input"
    onChange={[Function]}
    placeholder="search your chicken"
    type="search"
  />
</div>
`;
